rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -248,7 +248,21 @@ void SqliteStorage::write_locked(
             "' has not been created yet! Call 'create_topic' first.");
   }
 
-  write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
+  try {
+    write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
+  } catch (const SqliteException & exc) {
+    if (SQLITE_TOOBIG == exc.get_sqlite_return_code()) {
+      ROSBAG2_STORAGE_DEFAULT_PLUGINS_LOG_WARN_STREAM(
+        "Message on topic '" << message->topic_name << "' of size '"
+        << message->serialized_data->buffer_length
+        << "' bytes failed to write because it exceeds the maximum size sqlite can store: "
```

Comment:
Could we also include the SQLITE_MAX_LENGTH (https://www.sqlite.org/limits.html) value in this error message, so that the users can know what it's set at?

Reply:
So, this value doesn't appear to be in the headers from sqlite (it's a compiler definition, but it isn't stored in a header AFAIK). I can get it dynamically by "changing" the limit too `-1` and it will return the current value without modifying it, but this requires an sqlite database and I'm not sure I have one in scope here. I'll try.

Reply:
Ah, that's too bad. I don't think it's really critical, just a nice thing to know when it happens without needing to go to the documentation.

Reply:
I got it, I forgot we were in a method of the class that holds the db.